### PR TITLE
Support trigger validation on sub-folder

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#398386](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/398386)

### CODE REVIEW

File: src/build/buildController.ts [hunk 1]
```diff
@@ -52,10 +52,11 @@ export class BuildController implements Disposable {
         return this._instanceAvailable;
     }
 
-    public async build(correlationId: string): Promise<void> {
+    public async build(correlationId: string, buildWorkspace?: Uri): Promise<void> {
         let buildInput: BuildInput;
         this._eventStream.post(new BuildTriggered(correlationId, !!this._credentialController.credential.userInfo));
         const start = Date.now();
+        const buildType = !buildWorkspace || (buildWorkspace.fsPath.toLowerCase() == this._repositoryRoot.toLowerCase()) ? BuildType.FullBuild : BuildType.PartialBuild;
```
Comment (shanluo911) [inline src/build/buildController.ts]: nit: even strings here are not null. using '===' is still the best practice

Comment (928PJY) [inline src/build/buildController.ts]:
```suggestion
        const buildType = !buildWorkspace || (buildWorkspace.fsPath.toLowerCase() === this._repositoryRoot.toLowerCase()) ? BuildType.FullBuild : BuildType.PartialBuild;

```

**Review by shanluo911 — COMMENTED** [review]
should we update readme? or we can update it in release commit.

**Review by shanluo911 — APPROVED** [review]

**Comment by 928PJY** [comment]
> should we update readme? or we can update it in release commit.

Good point, I will update the readme in the release commit

**Pull request merged**